pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -1149,7 +1205,7 @@ func (r *OnlineUpgradeReconciler) removeReplicaGroupA(ctx context.Context) (ctrl
 	if verrors.IsReconcileAborted(res, err) {
```

Comment:
At the beginning of this function, add the condition: 
if vmeta.GetOnlineUpgradeStepInx(r.VDB.Annotations) > removeReplicaGroupA {
		return ctrl.Result{}, nil
}
It can prevent the index to be larger than 4.